src/itertools.ts

```diff
@@ -201,9 +178,14 @@ export function* islice<T>(
     stop = stopOrStart;
   }
 
-  const pred = slicePredicate(start, stop, step);
+  if (start < 0) throw new Error("start cannot be negative");
+  if (stop !== null && stop < 0) throw new Error("stop cannot be negative");
+  if (step <= 0) throw new Error("step cannot be negative");
+
   for (const [i, value] of enumerate(iterable)) {
-    if (pred(i)) {
+    if (i < start) continue;
+    if (stop !== null && i >= stop) break;
+    if (step === 1 || (i - start) % step === 0) {
```

Comment:
step === 1 is redundant as `n % 1` is always 0

Reply:
True, fixed in fe1f32c0caf06939222c9b2c18b42bcbdef6f751.